src/app/editor/example-contracts.js

```diff
@@ -1,96 +1,263 @@
-        sender.vote = toProposal;
-        proposals[toProposal].voteCount += sender.weight;
+        sender.vote = proposal;
+
+        // If 'proposal' is out of the range of the array,
+        // this will throw automatically and revert all
+        // changes.
+        proposals[proposal].voteCount += sender.weight;
     }
 
-    function winningProposal() public view returns (uint8 _winningProposal) {
-        uint256 winningVoteCount = 0;
-        for (uint8 prop = 0; prop < proposals.length; prop++)
-            if (proposals[prop].voteCount > winningVoteCount) {
-                winningVoteCount = proposals[prop].voteCount;
-                _winningProposal = prop;
+    /** 
+     * @dev Computes the winning proposal taking all previous votes into account.
+     * @return winningProposal_ index of winning proposal in the proposals array
+     */
+     
+    function winningProposal() public view
+            returns (uint winningProposal_)
+    {
+        uint winningVoteCount = 0;
+        for (uint p = 0; p < proposals.length; p++) {
+            if (proposals[p].voteCount > winningVoteCount) {
+                winningVoteCount = proposals[p].voteCount;
+                winningProposal_ = p;
             }
+        }
+    }
+
+    /** 
+     * @dev Calls winningProposal() function to get the index of the winner contained in the proposals array and then
+     * @return winnerName_ the name of the winner
+     */
+     
+    function winnerName() public view
+            returns (bytes32 winnerName_)
+    {
+        winnerName_ = proposals[winningProposal()].name;
     }
 }
 `
 
-var ballotTest = `
-pragma solidity >=0.4.22 <0.6.0;
+var ballotTest = `pragma solidity >=0.4.22 <0.7.0;
 import "remix_tests.sol"; // this import is automatically injected by Remix.
-import "./ballot.sol";
+import "./3_Ballot.sol";
 
-contract test3 {
+contract BallotTest {
+   
+    bytes32[] proposalNames;
    
     Ballot ballotToTest;
     function beforeAll () public {
-       ballotToTest = new Ballot(2);
+        proposalNames.push(bytes32("candidate1"));
+        ballotToTest = new Ballot(proposalNames);
     }
     
     function checkWinningProposal () public {
-        ballotToTest.vote(1);
-        Assert.equal(ballotToTest.winningProposal(), uint(1), "1 should be the winning proposal");
+        ballotToTest.vote(0);
+        Assert.equal(ballotToTest.winningProposal(), uint(0), "proposal at index 0 should be the winning proposal");
+        Assert.equal(ballotToTest.winnerName(), bytes32("candidate1"), "candidate1 should be the winner name");
     }
     
     function checkWinninProposalWithReturnValue () public view returns (bool) {
-        return ballotToTest.winningProposal() == 1;
+        return ballotToTest.winningProposal() == 0;
     }
 }
 `
 
 module.exports = {
-  ballot: { name: 'ballot.sol', content: ballot },
-  ballot_test: { name: 'ballot_test.sol', content: ballotTest }
+  storage: { name: '1_Storage.sol', content: storage },
```

Comment:
do we really need to prefix with "1_"

Reply:
I asked him to add it to relfect the level of difficulty. It helps beginner to know where to start

Reply:
perhaps should we write directly in the code the level of difficulty.
that's not really obvious that "x_" means that

Reply:
Ok, I thought it was more obvious with a prefix. But a comment in the code is good for me too.